src/d/actor/d_a_obj_hat.cpp

```diff
@@ -4,91 +4,238 @@
+    // if (this->actor_condition & )
+    fopAcM_SetupActor(this, daObjHat_c);
+
+    int var = dComIfG_resLoad(&mPhs, "Ro");
+
+    if (var == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap((fopAc_ac_c*)this, &CheckCreateHeap, 0uL)) {
+            return cPhs_ERROR_e;
+        } else {
+            return createInit();
+        }
+    }
+
     /* Nonmatching */
 }
 
+static const int l_bmd_ix_tbl[] = {RO_BDL_RO_HAT, RO_BDL_RO_HAT2, RO_BDL_RO_HAT3, RO_BDL_RO_HAT};
+static const int l_bck_ix_tbl[] = {RO_INDEX_BCK_HAT_WID, RO_BCK_HAT2_WIND, RO_BCK_HAT3_WID,
+                                   RO_BCK_HAT_WID};
+static dCcD_SrcCyl l_cyl_src = {
+    // dCcD_SrcGObjInf
+    {
+        /* Flags             */ 0,
+        /* SrcObjAt  Type    */ 0,//AT_TYPE_UNK1000,
+        /* SrcObjAt  Atp     */ 0,
+        /* SrcObjAt  SPrm    */ 0,
+        /* SrcObjTg  Type    */ AT_TYPE_WIND,
+        /* SrcObjTg  SPrm    */ cCcD_TgSPrm_Set_e | cCcD_TgSPrm_IsEnemy_e,
+        /* SrcObjCo  SPrm    */ cCcD_CoSPrm_Set_e | cCcD_CoSPrm_IsOther_e | cCcD_CoSPrm_VsGrpAll_e,
+        /* SrcGObjAt Se      */ 0,
+        /* SrcGObjAt HitMark */ 0,
+        /* SrcGObjAt Spl     */ 0,
+        /* SrcGObjAt Mtrl    */ 0,
+        /* SrcGObjAt SPrm    */ 0,//AT_TYPE_WIND,
+        /* SrcGObjTg Se      */ 0,
+        /* SrcGObjTg HitMark */ 0,
+        /* SrcGObjTg Spl     */ 0,
+        /* SrcGObjTg Mtrl    */ 0,
+        /* SrcGObjTg SPrm    */ dCcG_TgSPrm_NoHitMark_e | dCcG_TgSPrm_NoConHit_e,
+        /* SrcGObjCo SPrm    */ 0,
+    },
+    // cM3dGCylS
+    {
+        /* Center */ 0.0f,
+        0.0f,
+        0.0f,
+        /* Radius */ 70.0f,
+        /* Height */ 80.0f,
+    },
+};
+
 /* 0000045C-000005D4       .text createHeap__10daObjHat_cFv */
-void daObjHat_c::createHeap() {
-    /* Nonmatching */
+BOOL daObjHat_c::createHeap() {
+    J3DModelData* pModelData = (J3DModelData*)dComIfG_getObjectIDRes("Ro", l_bmd_ix_tbl[this->id]);
+    if (pModelData == NULL) {
+        return FALSE;
+    } else {
+        mDoExt_McaMorf* morf = new mDoExt_McaMorf(
+            pModelData, NULL, NULL,
+            (J3DAnmTransformKey*)dComIfG_getObjectIDRes("Ro", l_bck_ix_tbl[this->id]),
+            J3DFrameCtrl::LOOP_REPEAT_e, 1.0f, 0, -1, 1, NULL, 0x80000, 0x37441422);
+
+        this->mpMorf = morf;
+        if (this->mpMorf == NULL || this->mpMorf->getModel() == NULL) {
+            return FALSE;
+        } else {
+            this->model = this->mpMorf->getModel();
+            acch_cir.SetWall(30.0f, 30.0f);
+            acch.Set(&this->current.pos, &this->old.pos, this, 1, &acch_cir, &this->speed,
+                     &this->current.angle, &this->shape_angle);
+            return TRUE;
+        }
+    }
+    return FALSE;
 }
 
 /* 000005D4-000006AC       .text createInit__10daObjHat_cFv */
-void daObjHat_c::createInit() {
-    /* Nonmatching */
+s32 daObjHat_c::createInit() {
+    stts.Init(2, 0xff, this);
+    cyl.Set(l_cyl_src);
+    this->cyl.SetStts(&this->stts);
```

Comment:
```suggestion
    mCyl.SetStts(&mStts);
```
Typing `this->` is unnecessary. This also applies everywhere else in this file.